lib/matplotlib/pyplot.py

```diff
@@ -1146,6 +1159,23 @@ def subplots(nrows=1, ncols=1, sharex=False, sharey=False, squeeze=True,
                        gridspec_kw=gridspec_kw)
     return fig, axs
 
+def subplots_iterator(nrows=1, ncols=1, show_in_between=False, sharex=False,
+                      sharey=False, squeeze=True, subplot_kw=None,
+                      gridspec_kw=None, **fig_kw):
+    """ Iteratively yields the axis object of a rows x cols subplot and creates new subplots when needed"""
+    for axis in (
```

Comment:
Yes, Axes is the plural of axis. This has been a long-standing problem for those of us who write documentation, tutorials and books about matplotlib. Thomas is right, though... you are iterating over Axes objects, not Axis objects (an Axes object is made of 2 or 3 Axis objects, among other things).

So, the documentation needs to say ```:class:`Axes` ```, and code that iterates over a list of them typically uses `ax` to refer to a single one of them, not `axis`.